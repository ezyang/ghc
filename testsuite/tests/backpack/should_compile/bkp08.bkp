package q where
    module H where
        data T = T { x :: Bool }
package r where
    signature H where
        data T
package p where
    include q
    include r
    module M where
        import H
        f = T True
