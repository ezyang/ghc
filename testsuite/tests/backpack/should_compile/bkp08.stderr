==== Package q ====
Shape for q
provides: H -> THIS:H
               T{T, T, x}
requires:
[1 of 1] Compiling H                ( q/H.hs, bkp08-out/Br2hqQttvyiBkpnTZJaMkb/H.o )
==== Package r ====
Shape for r
provides:
requires: H -> THIS:H
               T{T}
[1 of 1] Compiling H[abstract sig]  ( r/H.hsig, nothing )
==== Package p ====
Shape for p
provides: H -> q():H
               T{T, T, x}
          M -> THIS:M
               f
requires:
[1 of 3] Including q
==== Package q ====
Shape for q
provides: H -> THIS:H
               T{T, T, x}
requires:
[2 of 3] Including r
==== Package r ====
Shape for r
provides:
requires: H -> THIS:H
               T{T}
[1 of 1] Compiling H[abstract sig]  ( r/H.hsig, nothing )
[3 of 3] Compiling M                ( p/M.hs, bkp08-out/3TzWjyLYxpKKUrnKgPCj8v/M.o )
